scipy/optimize/_linprog_ip.py

```diff
@@ -114,7 +114,10 @@ def solve(r):
                 # this seems to cache the matrix factorization, so solving
                 # with multiple right hand sides is much faster
                 def solve(r, sym_pos=sym_pos):
-                    return sp.linalg.solve(M, r, sym_pos=sym_pos)
+                    if sym_pos:
+                        return sp.linalg.solve(M, r, assume_a="pos")
+                    else:
+                        return sp.linalg.solve(M, r)
```

Comment:
Not 100% sure I understand completely how this function (return) is exposed - `_get_solver` only seems to be used in this file, but I think we should very likely change the signature from `sym_pos` to `assume_a` here as well...?

Reply:
That solver returns in an internal result and `sym_pos` is a kwarg of the optimization routine. As is it will do the correct thing.

Reply:
I'm not doubting the correctness of the code as-is, I'm wondering if the APIs should be aligned, even if the API is not consumed publicly. The translation from `assume_a` to `sym_pos` - if still necessary due to handing over due to a function that's not "owned" by scipy - could happen at that interface, for example.

Reply:
True but the optimization algorithm doesn't have any other options like triangular or symmetric etc. like solve has. Hence it provides only a single alternative that might be served best with a bool. `linalg.solve` will (hopefully) start auto discovering structure in the next version anyways so we won't be needing these.